frontend/src/components/UpdateThread/UpdateThreadForm.js

```diff
@@ -0,0 +1,110 @@
+import React, { useEffect, useState } from 'react';
+import axios from 'axios';
+import { Form, Modal, Divider } from 'antd';
+import { useNavigate } from 'react-router-dom';
+import CreateThreadContent from '../CreateThread/CreateThreadContent'
+import '../../style/CreateThread/CreateThreadForm.css'
+import CreateThreadPetInfo from '../CreateThread/CreateThreadPetInfo'
+import useThreadTypeKeywordSwitch from '../CreateThread/useThreadTypeKeywordSwitch'
+import { useDispatch, useSelector} from 'react-redux';
+import { updateThreadAsync, getThreadAsync} from '../../thunk/threadThunk';
+import moment from 'moment';
+
+function UpdateThreadForm ({ open, onUpdate, onCancel, threadId }) {
+  const [threadType, updateThreadType] = useState('');
+  const navigate = useNavigate();
+  const dispatch = useDispatch();
+  const user = useSelector(state => state.user);
+  const [form] = Form.useForm();
+
+  useEffect(() => {
+    dispatch(getThreadAsync(threadId)).then((res) => {
+      console.log("Got thread data: ", res);
+      const thread = res.payload.thread;
+      console.log('thread from update: ', thread);
+      // if (!thread.pet || !thread.kind) return; // some checks, shouldn't be triggered at all
+
+      axios
+        .get(`http://localhost:3001/pet/${thread.pet}`)
```

Comment:
for a thread to be displayed, clicked and entered, it must have been fetched and stored in the forumSlice, so maybe you can directly get the thread from the redux state instead of getting it again calling the api.

Reply:
I'll take a look and change it in the next version I guess. But not for this PR. @ddm1293